crates/ln-dlc-node/src/channel.rs

```diff
@@ -0,0 +1,189 @@
+use anyhow::bail;
+use anyhow::Result;
+use bitcoin::hashes::hex::ToHex;
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::Txid;
+use dlc_manager::ChannelId;
+use lightning::ln::channelmanager::ChannelDetails;
+use lightning::util::events::ClosureReason;
+use std::fmt;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::str::FromStr;
+use time::OffsetDateTime;
+use uuid::Uuid;
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum ChannelState {
+    Pending,
+    Open,
+    Closed,
+    ForceClosedRemote,
+    ForceClosedLocal,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct UserChannelId(Uuid);
+
+impl Default for UserChannelId {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl UserChannelId {
+    pub fn new() -> UserChannelId {
+        UserChannelId(Uuid::new_v4())
+    }
+
+    pub fn to_u128(&self) -> u128 {
+        self.0.as_u128()
+    }
+}
+
+impl ToString for UserChannelId {
+    fn to_string(&self) -> String {
+        self.0.to_string()
+    }
+}
```

Comment:
🔧 From the [`ToString` docs](https://doc.rust-lang.org/std/string/trait.ToString.html):

> This trait is automatically implemented for any type which implements the [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html) trait. As such, ToString shouldn’t be implemented directly: [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html) should be implemented instead, and you get the ToString implementation for free.